src/interpret.ts

```diff
@@ -339,17 +339,26 @@ function timeout(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-interface ResultIndexItem {
+export interface SuccessIndexItem {
```

Comment:
I think it's worth adding a `type` or `t` property here (or whatever convention is used elsewhere) to explicitly call out whether something is a success or a failure.

At the minute, in the VS Code extension code, we rely on the presence of an `error` property which works but isn't great and makes our types a bit confusing.

Thoughts @rneatherway?

Reply:
Yes, I agree. I found this in the older version of the TypeScript Handbook: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions, which suggests the same. I'm a bit disappointed that there isn't a type-level `switch` but I guess you can't win them all.

Reply:
Changed the shape of the result index in https://github.com/dsp-testing/qc-run2/pull/567/commits/aadf5bb7edd44fb10e1af30aa167fcbe47806d82 (and tested with corresponding changes in the VS Code extension). Seems to work 🎉